packages/swc/src/module.ts

```diff
@@ -0,0 +1,35 @@
+import type { Plugin } from 'rollup';
+import { transform } from '@swc/core';
+import { merge } from 'smob';
+
+import type { Options } from './type';
+
+export function swc(input: Options = {}): Plugin {
+  const options: Options = merge(
+    {
+      jsc: {
+        target: 'es2020',
+        parser: {
+          syntax: 'typescript',
+          decorators: true
+        },
+        transform: {
+          decoratorMetadata: true,
+          legacyDecorator: true
+        },
+        loose: true
+      }
+    },
+    input
+  );
+
+  return {
+    name: 'swc',
+    transform(code) {
+      return transform(code, {
+        ...options,
+        sourceMaps: true
```

Comment:
I’d have to check but doesn’t rollup provide a signal as to whether source maps are requested?

Reply:
rollup requires, as far as I could verify, that all transformers provide a sourcemap. Only plugins that use e.g. the renderChunk hook are supposed to provide a sourcemap depending on the corresponding value of the OutputOptions parameter.
[https://rollupjs.org/plugin-development/#renderchunk](https://rollupjs.org/plugin-development/#renderchunk)